[Demo] Add radar chart to visual-vocabulary

rc678 opened the pull request:
## Description

Added a radar chart to the visual vocabulary.

Resources:

- https://plotly.com/python/radar-chart/
- https://plotly.com/python-api-reference/generated/plotly.express.line_polar
- https://en.wikipedia.org/wiki/Radar_chart


## Screenshot

Gallery page:

![image](https://github.com/user-attachments/assets/97de3d7f-9f7b-48e3-861e-fddd9226a794)


Radar chart page:

![image](https://github.com/user-attachments/assets/af0749fe-d34c-48fd-9a19-30e55ccfcf11)


## Notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/README.md:
```diff
@@ -81,7 +81,7 @@ The dashboard is still in development. Below is an overview of the chart types f
 | Pictogram             | ❌     | Magnitude                |                                                                                            |                                                                                                                                                                                                 |
 | Pie                   | ✅     | Part-to-whole            | [Pie chart with px](https://plotly.com/python/pie-charts/)                                 | [px.pie](https://plotly.com/python-api-reference/generated/plotly.express.pie)                                                                                                                  |
-| Radar                 | ❌     | Magnitude                |                                                                                            |                                                                                                                                                                                                 |
+| Radar                 | ✅     | Magnitude                |                                                                                            |                                                                                                                                                                                                 |
```
If you could add these links (see other charts), then that would be great!

```
[Radar chart with px](https://plotly.com/python/radar-chart/)
[px.line_polar](https://plotly.com/python-api-reference/generated/plotly.express.line_polar)
```

rc678 replied:
Added links

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/examples/radar.py:
```diff
@@ -0,0 +1,15 @@
+import vizro.models as vm
+import vizro.plotly.express as px
+from vizro import Vizro
+
+wind = px.data.wind()
+
+page = vm.Page(
+    title="Radar chart",
+    components=[
+        vm.Graph(figure=px.line_polar(wind, r="frequency", theta="direction", color="strength", line_close=True))
+    ],
+)
+
+dashboard = vm.Dashboard(pages=[page])
+Vizro().build(dashboard).run()
```
This example was already beautiful! I've made a few minor tweaks, mainly to simplify the chart. While the `color `dimension offers an interesting use-case, I recommend starting with just one category to keep the introduction of the chart straightforward.

I've also included a custom chart below to create the classic "radar" chart with a filled area. This custom chart is necessary so that we can utilize a post-update call - see our docs: https://vizro.readthedocs.io/en/stable/pages/user-guides/custom-charts/

Simply add the custom chart code to `custom_charts.py` and add some docstrings (refer to the butterfly example for guidance).

```suggestion
import vizro.models as vm
import vizro.plotly.express as px
from vizro import Vizro
from vizro.models.types import capture

wind = px.data.wind()


@capture("graph")
def radar(data_frame, **kwargs):
    """
    Add docstrings here (see butterfly example)
    
    """
    fig = px.line_polar(data_frame, **kwargs)
    fig.update_traces(fill="toself")
    return fig


page = vm.Page(
    title="Radar chart",
    components=[
        vm.Graph(
            figure=radar_chart(
                wind.query("strength =='1-2'"), r="frequency", theta="direction", color="strength", line_close=True
            )
        )
    ],
)

dashboard = vm.Dashboard(pages=[page])
Vizro().build(dashboard).run()
```

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/magnitude.py:
```diff
@@ -190,4 +190,31 @@
+                as well as identifying patterns or outliers.
+        """
+        ),
+        vm.Graph(figure=px.line_polar(wind, r="frequency", theta="direction", color="strength", line_close=True)),
```
```suggestion
        vm.Graph(figure=radar(wind.query("strength =='1-2'"), r="frequency", theta="direction", line_close=True)
        ),
```

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/magnitude.py:
```diff
@@ -3,7 +3,7 @@
 
-from pages._pages_utils import PAGE_GRID, gapminder, iris, make_code_clipboard_from_py_file, tips
+from pages._pages_utils import PAGE_GRID, gapminder, iris, make_code_clipboard_from_py_file, tips, wind
 
```
```suggestion
from custom_charts import radar
```

huong-li-nguyen replied:
For this to work, you need to add the below code to the `custom_charts.py` - please fill out the docstrings :) 

```
@capture("graph")
def radar(data_frame, **kwargs):
    """
    Add docstrings here (see butterfly example)
    
    """
    fig = px.line_polar(data_frame, **kwargs)
    fig.update_traces(fill="toself")
    return fig
```

rc678 replied:
Done

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/magnitude.py:
```diff
@@ -190,4 +190,31 @@
+            text="""
+                #### What is a radar chart?
+
+                A radar chart is also known as a spider plot or star plot.
+                It is a type of data visualization used to display multivariate data.
+                The chart has three or more variables represented on axes with
+                the same starting point.
```
```suggestion
                A radar chart, also known as a spider plot or star plot, is a type of data visualization used to display multivariate data. It consists of three or more variables represented on axes that originate from the same central point.
```

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/magnitude.py:
```diff
@@ -190,4 +190,31 @@
+
+                #### When should I use it?
+
+                Use a radar chart to visualize and compare the performance or
+                characteristics of data across multiple variables.
+                This chart is useful for highlighting strengths and weaknesses,
+                as well as identifying patterns or outliers.
```
```suggestion
                Use a radar chart to compare performance or characteristics across multiple variables. It effectively highlights strengths, weaknesses, patterns, and outliers.

For clarity, limit the number of variables and data series. Use consistent scales for all axes, clearly mark labels and data points, and avoid clutter by focusing on key variables.
```

huong-li-nguyen reviewed (commented):
Hello @rc678 ,

congratulations on your first PR in this repository! Even though you mentioned in your PR title to not review yet, I thought I'll give you early feedback and suggestions as this PR is almost ready to merge for me :) 

I've added a few suggestions mostly to simplify the chart for now. The code and the chart already looked great though! ⭐

rc678 commented:
> Hello @rc678 ,
> 
> congratulations on your first PR in this repository! Even though you mentioned in your PR title to not review yet, I thought I'll give you early feedback and suggestions as this PR is almost ready to merge for me :)
> 
> I've added a few suggestions mostly to simplify the chart for now. The code and the chart already looked great though! ⭐

@huong-li-nguyen Thank you for all the feedback! I've addressed all the comments and I am ready for a re-review.

huong-li-nguyen commented on vizro-core/examples/visual-vocabulary/pages/examples/radar.py:
```diff
@@ -0,0 +1,29 @@
+    components=[
+        vm.Graph(
+            figure=radar(
+                wind.query("strength =='1-2'"), r="frequency", theta="direction", color="strength", line_close=True
```
```suggestion
                wind.query("strength =='1-2'"), r="frequency", theta="direction", line_close=True
```

huong-li-nguyen reviewed (approved):
Perfect - thanks for incorporating the PR comments so quickly! This looks great! 🌟

antonymilne commented on vizro-core/examples/visual-vocabulary/pages/examples/radar.py:
```diff
@@ -0,0 +1,25 @@
+page = vm.Page(
+    title="Radar",
+    components=[
+        vm.Graph(figure=radar(wind.query("strength =='1-2'"), r="frequency", theta="direction", line_close=True))
```
```suggestion
        vm.Graph(figure=radar(wind.query("strength == '1-2'"), r="frequency", theta="direction", line_close=True))
```

antonymilne commented on vizro-core/examples/visual-vocabulary/pages/magnitude.py:
```diff
@@ -190,4 +191,27 @@
+                To maintain clarity, use consistent scales for all axes and clearly mark labels and data points.
+        """
+        ),
+        vm.Graph(figure=radar(wind.query("strength =='1-2'"), r="frequency", theta="direction", line_close=True)),
```
```suggestion
        vm.Graph(figure=radar(wind.query("strength == '1-2'"), r="frequency", theta="direction", line_close=True)),
```

antonymilne reviewed (approved):
This is amazing, thank you very much for your contribution @rc678! 🙏

@huong-li-nguyen the only thing we might like to do is add a cross-links between/references this page and parallel coordinates, given they're basically the same but radial instead of parallel. I don't know whether we do any other cross-linking/reference on this dashboard though, so maybe it would be weird to put it here. So up to you.

huong-li-nguyen commented:
> @huong-li-nguyen the only thing we might like to do is add a cross-links between/references this page and parallel coordinates, given they're basically the same but radial instead of parallel. I don't know whether we do any other cross-linking/reference on this dashboard though, so maybe it would be weird to put it here. So up to you.

Let's do it in a separate PR in another iteration, because we don't do any cross-links so far 👍
